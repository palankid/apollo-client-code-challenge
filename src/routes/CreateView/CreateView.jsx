import React from 'react';
import { useHistory } from 'react-router-dom';
import { useMutation, gql } from '@apollo/client';

import BookOperationsForm from '../../components/BookOperationsForm';
import { routeNames } from '../../config/routes.config';

const CREATE_BOOK = gql`
    mutation CreateBook($title: String!, $author: String!, $price: Float!) {
        createBook(title: $title, author: $author, price: $price) {
            bookId,
            title,
            author,
            price
        }
    }
`;

const CreateView = () => {
    const history = useHistory();
    const [createBook, { data }] = useMutation(CREATE_BOOK);

    const handleFinish = variables => {
        createBook({ variables });
    };

    const handleCancel = () => {
        history.push(routeNames.root);
    }

    return (
        <BookOperationsForm
            isCreateMode={true}
            onFinish={handleFinish}
            onCancel={handleCancel}
        />
    )
}

export default CreateView;
