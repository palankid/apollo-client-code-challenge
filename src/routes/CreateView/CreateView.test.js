import React from 'react';
import { shallow } from 'enzyme';

import CreateView from './CreateView';

jest.mock('./operations/mutations/createBook', () => ({
    useCreateBook: () => ({
        loading: false,
        error: null,
        handleFinish: jest.fn(),
        handleCancel: jest.fn()
    })
}));

describe('CreateView', () => {
    it('should test that component is rendered and not changed unintentionally', () => {
        const wrapper = shallow(
            <CreateView />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
