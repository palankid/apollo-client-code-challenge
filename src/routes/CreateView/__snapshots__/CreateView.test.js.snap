// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateView should test that component is rendered and not changed unintentionally 1`] = `
<div
  className="create-view"
>
  <BookOperationsHeader
    titleFragment="Create"
  />
  <BookOperationsForm
    disabled={false}
    initialValues={
      Object {
        "author": "",
        "id": null,
        "price": null,
        "title": "",
      }
    }
    isCreateMode={true}
    loading={false}
    onCancel={[MockFunction]}
    onFinish={[MockFunction]}
  />
</div>
`;
